altair/utils/data.py

```diff
@@ -140,11 +136,30 @@ def check_data_type(data):
 # Private utilities
 # ==============================================================================
 
+def _compute_data_hash(data_str):
+    return hashlib.md5(data_str.encode()).hexdigest()
 
-def _compute_uuid_filename(prefix, ext):
-    return prefix + '-' + str(uuid.uuid4()) + ext
+
+def _data_to_json_string(data):
+    """Return a JSON string representation of the input data"""
+    check_data_type(data)
+    if isinstance(data, pd.DataFrame):
+        data = sanitize_dataframe(data)
+        return data.to_json(orient='records')
+    elif isinstance(data, dict):
+        if 'values' not in data:
+            raise KeyError('values expected in data dict, but not present.')
+        return json.dumps(data['values'], sort_keys=True)
+    else:
+        raise NotImplementedError("to_json only works with data expressed as "
+                                  "a DataFrame or as a dict")
 
 
-def _compute_filename(prefix='altair-data', ext='.csv'):
-    filename = _compute_uuid_filename(prefix, ext)
-    return filename
+def _data_to_csv_string(data):
+    """return a CSV string representation of the input data"""
+    check_data_type(data)
+    if isinstance(data, pd.DataFrame):
```

Comment:
Just thinking - shouldn't the csv transformer also work with dict/values as the json transformer does? I know the original code didn't work this way, but I don't see any reason to not add the same logic here.

Reply:
Sounds good – I'll add that.